### PR TITLE
[Cleanup] Remove unused `to_postgres_type` function from codegen

### PR DESCRIPTION
**Pull request by JasoonS**

### CODE REVIEW

File: codegenerator/cli/src/config_parsing/entity_parsing.rs [hunk 1]
```diff
@@ -1266,24 +1243,6 @@ impl GqlScalar {
             name => GqlScalar::Custom(name.to_string()),
         }
     }
-    fn to_postgres_type(&self, schema: &Schema) -> anyhow::Result<String> {
```
Comment (JasoonS) [inline codegenerator/cli/src/config_parsing/entity_parsing.rs:1269]: This function is completely unused, so I removed all references to it and everything is still working correctly.

File: codegenerator/cli/src/config_parsing/entity_parsing.rs [hunk 2]
```diff
@@ -1767,115 +1745,112 @@ type TestEntity {
...
+            .expect("Failed to get postgres field")
+            .unwrap();
+
+        assert_eq!(pg_field.field_name, "status");
+        assert_eq!(pg_field.field_type, PGPrimitive::Enum("Status".to_string()));
+        assert!(!pg_field.is_index);
+        assert!(!pg_field.is_array);
+        assert!(!pg_field.is_nullable);
+        assert_eq!(pg_field.linked_entity, None);
+    }
```
Comment (JasoonS) [inline codegenerator/cli/src/config_parsing/entity_parsing.rs:1855]: These were generated by ChatGPT but they check out and work pretty well :blush:

File: codegenerator/cli/src/config_parsing/entity_parsing.rs [hunk 3]
```diff
@@ -1682,57 +1644,73 @@ type TestEntity {
...
-        gql_type_to_postgres_type_test_helper(gql_type);
+    fn gql_multi_nullable_array_to_pgprimitive_should_panic() {
+        let gql_type = "[[Int!]]!"; // Nested lists need to be not nullable
+        let field_type = get_field_type_helper(gql_type);
+        let empty_schema = Schema::empty();
+        let _pg_primitive = field_type
+            .to_user_defined_field_type()
+            .to_underlying_postgres_primitive(&empty_schema)
+            .expect("should panic due to validation error");
     }
```
Comment (JasoonS) [inline codegenerator/cli/src/config_parsing/entity_parsing.rs:1714]: I had to update these old tests to work on the relevant code.

File: codegenerator/cli/src/hbs_templating/codegen_templates.rs [hunk 4]
```diff
@@ -149,7 +149,6 @@ pub struct EntityParamTypeTemplate {
     pub field_name: CapitalizedOptions,
     pub res_type: RescriptTypeIdent,
     pub res_schema_code: String,
-    pub type_pg: String,
```
Comment (JasoonS) [inline codegenerator/cli/src/hbs_templating/codegen_templates.rs:152]: This was not being used in the Handlebars templates.

**Review by DZakh — APPROVED** [review]
Nice 🚀

**Pull request merged**